data-prepper-api/src/test/java/org/opensearch/dataprepper/model/buffer/AbstractBufferTest.java

```diff
@@ -198,6 +212,22 @@ public void testWriteAllTimeoutMetric() throws TimeoutException {
         Assert.assertEquals(1.0, timeoutMeasurements.get(0).getValue(), 0);
     }
 
+    @Test
+    public void testWriteAllRecordsOverflowMetric() {
+        // Given
+        final AbstractBuffer<Record<String>> abstractBuffer = new AbstractBufferRuntimeExceptionImpl(testPluginSetting);
```

Comment:
You could accomplish this via mocking:

```
final AbstractBuffer<Record<String>> abstractBuffer = mock(AbstractBuffer.class);
when(abstractBuffer).doWriteAll(any(Collection.class), anyInteger()).thenThrow(RuntimeException.class);
```

Then you can remove the `AbstractBufferRuntimeExceptionImpl` class.

Reply:
I see what you mean. For future reference, you can configure Mockito to call real methods.

```
when(abstractBuffer.writeAll(any(Collection.class), anyInteger()).thenCallRealMethod();
```